src/transformers/models/llava_next/modeling_llava_next.py

```
@@ -0,0 +1,684 @@
+                            self.config.vision_config.image_size,
+                        )
+                        image_feature = image_feature.view(num_patch_height, num_patch_width, height, width, -1)
+                        image_feature = image_feature.permute(4, 0, 2, 1, 3).contiguous()
+                        image_feature = image_feature.flatten(1, 2).flatten(2, 3)
+                        image_feature = unpad_image(image_feature, image_sizes[image_idx])
+                        image_feature = torch.cat(
+                            (
+                                image_feature,
+                                self.image_newline[:, None, None].expand(*image_feature.shape[:-1], 1),
+                            ),
+                            dim=-1,
+                        )
+                        image_feature = image_feature.flatten(1, 2).transpose(0, 1)
+                        image_feature = torch.cat((base_image_feature, image_feature), dim=0)
+                    else:
+                        image_feature = image_feature[0]
+                        image_feature = torch.cat((image_feature, self.image_newline[None]), dim=0)
+                    new_image_features.append(image_feature)
+                image_features = torch.stack(new_image_features, dim=0)
+
+                inputs_embeds, attention_mask, labels, position_ids = self._merge_input_ids_with_image_features(
+                    image_features, inputs_embeds, input_ids, attention_mask, labels
+                )
+                if labels is None:
+                    labels = torch.full_like(attention_mask, self.config.ignore_index).to(torch.long)
+            else:
+                # In case input_ids.shape[1] == 1 & pixel_values==None & past_key_values != None, we are in the case of
+                # generation with cache
+                if past_key_values is not None and pixel_values is not None and input_ids.shape[1] == 1:
```

What if any of these conditions aren't true - what situation are we in? If it's not valid, we should raise an error. For example, can I call this method directly with `input_embeds` if `input_ids` is None?

```suggestion
            elif past_key_values is not None and pixel_values is not None and input_ids.shape[1] == 1:
                # In case input_ids.shape[1] == 1 & pixel_values==None & past_key_values != None, we are in the case of
                # generation with cache
```